rca/programmes/models.py

```diff
@@ -819,6 +830,29 @@ def get_programme_stories(self, programme_stories):
             "slides": related_list_block_slideshow(programme_stories.slides),
         }
 
+    def get_related_staff(self):
+        """Method to return a related staff.
+        The default behaviour should be to find related staff via the relationship
+        StaffPage.roles. This also needs to offer the option to
+        manually add related staff to the programme page, this helps solve issues
+        of custom ordering that's needed with large (>25) staff items.
+        """
+
+        from rca.people.models import StaffPage
+
+        related_staff = self.related_staff.select_related("image")
+        if related_staff:
+            return related_staff
+
+        # For any automatially related staff, adjust the list so we don't have
+        # to make edits to the template shared by other page models.
+        staff = []
+        for item in (
+            StaffPage.objects.filter(roles__programme=self).live().order_by("last_name")
+        ).distinct():
```

Comment:
Do/Why do we need to call distinct() here?

Reply:
When I tested this, I noticed that some staff had multiple roles specified for the same programme. For example:

```console
>>> from rca.people.models import StaffPage
>>> from rca.programmes.models import ProgrammePage
>>> ProgrammePage.objects.live().filter(title='Arts & Humanities')
<PageQuerySet [<ProgrammePage: Arts & Humanities MPhil/PhD>]>
>>> prog = ProgrammePage.objects.live().filter(title='Arts & Humanities').first()
>>> from pprint import pprint as pp
>>> pp(list(StaffPage.objects.filter(roles__programme=prog).live().order_by("last_name")), indent=2)
[ <StaffPage: Professor Gemma Blackshaw>,
  <StaffPage: Dr Nicky Coutts>,
  <StaffPage: Dr Chantal Faust>,
  <StaffPage: Professor Johnny Golding>,
  <StaffPage: Professor Johnny Golding>,
  <StaffPage: Professor Johnny Golding>,
  <StaffPage: Jonathan Miles>,
  <StaffPage: Dr Peter Oakley>]
>>> 
```

Professor Golding appears three times. Here's what the StaffPage looks like in Wagtail admin:

<details>
  <summary>screenshot</summary>

![image](https://github.com/torchbox/rca-wagtail-2019/assets/7713776/57c38e8e-8bf4-4516-b0a4-530f25a68ec3)

</details>

Reply:
ah yea that makes a lot of sense! thanks